app/javascript/components/flavor-form/add-form-schema.js

```diff
@@ -0,0 +1,163 @@
+      label: __('Provider'),
+      validateOnMount: true,
+      isSearchable: true,
+    },
+    {
+      component: componentTypes.TEXT_FIELD,
+      name: 'name',
+      validate: [{
+        type: validatorTypes.REQUIRED,
+        message: __('Required'),
+      }],
+      label: __('Name'),
+      validateOnMount: true,
+    },
+    {
+      component: componentTypes.TEXT_FIELD,
+      name: 'ram',
+      validate: [{
+        type: validatorTypes.REQUIRED,
+        message: __('Required'),
+      },
+      {
+        type: validatorTypes.PATTERN_VALIDATOR,
+        pattern: '^[-+]?[0-9]\\d*$',
+        message: __('Ram must be integer'),
+      },
+      {
+        type: validatorTypes.MIN_NUMBER_VALUE,
+        value: 1,
+        message: __('Ram must be greater than 0'),
+      },
+      ],
+      label: __('Ram size in MB'),
+      dataType: 'integer',
+      type: 'number',
+      validateOnMount: true,
+    },
+    {
+      component: componentTypes.TEXT_FIELD,
+      name: 'vcpus',
+      validate: [{
+        type: validatorTypes.REQUIRED,
+        message: __('Required'),
+      },
+      {
+        type: validatorTypes.PATTERN_VALIDATOR,
+        pattern: '^[-+]?[0-9]\\d*$',
+        message: __('VCPUs must be integer'),
+      },
+      {
+        type: validatorTypes.MIN_NUMBER_VALUE,
+        value: 1,
+        message: __('VCPUs must be greater than 0'),
+      }],
+      label: __('VCPUs'),
+      dataType: 'integer',
+      type: 'number',
+      validateOnMount: true,
+    },
+    {
+      component: componentTypes.TEXT_FIELD,
+      name: 'disk',
+      validate: [{
+        type: validatorTypes.REQUIRED,
+        message: __('Required'),
+      },
+      {
+        type: validatorTypes.PATTERN_VALIDATOR,
+        pattern: '^[-+]?[0-9]\\d*$',
+        message: __('Disk size must be integer'),
+      },
+      {
+        type: validatorTypes.MIN_NUMBER_VALUE,
+        value: 1,
+        message: __('Disk size must be greater than 0'),
+      }],
+      label: __('Disk size in GB'),
+      dataType: 'integer',
+      type: 'number',
+      validateOnMount: true,
+    },
+    {
+      component: componentTypes.TEXT_FIELD,
+      name: 'swap',
+      validate: [{
+        type: validatorTypes.REQUIRED,
+        message: __('Required'),
+      },
+      {
+        type: validatorTypes.PATTERN_VALIDATOR,
+        pattern: '^[-+]?[0-9]\\d*$',
+        message: __('Swap size must be integer'),
+      },
+      {
+        type: validatorTypes.MIN_NUMBER_VALUE,
+        value: 0,
+        message: __('Swap size must be greater or equal to 0'),
+      }],
+      label: __('Swap size in MB'),
+      dataType: 'integer',
+      type: 'number',
+      validateOnMount: true,
+    },
+    {
+      component: componentTypes.TEXT_FIELD,
+      name: 'rxtx_factor',
+      validate: [{
+        type: validatorTypes.REQUIRED,
+        message: __('Required'),
+      },
+      {
+        type: validatorTypes.PATTERN_VALIDATOR,
```

Comment:
I do not like the fact, that we are repeating the regexp for the number over and over again. Can you do something about it? (a follow up PR would be ok)